bundles/mapping/mapmodule/plugin/markers/MarkersPlugin.ol.js

```diff
@@ -148,315 +117,182 @@ Oskari.clazz.define('Oskari.mapframework.mapmodule.MarkersPlugin',
+                this.getMsg('title'),
+                this.getMsg('dialog.message'),
+                [clearBtn, closeBtn]
+            );
+            this.getMapModule().getMapEl().addClass(
+                'cursor-crosshair'
+            );
+            this.dialog.moveTo(
+                '#toolbar div.toolrow[tbgroup=default-selectiontools]',
+                'bottom'
+            );
         },
-
         /**
          * Stops the marker location selector
          */
         stopMarkerAdd: function (selectDefault) {
-            var me = this;
-            var sandbox = this.getSandbox();
-            me._waitingUserClickToAddMarker = false;
-            if (me.dialog) {
-                me.getMapModule().getMapEl().removeClass('cursor-crosshair');
-                me.dialog.close(true);
+            this.enableGfi(true);
+            this._showAddMarkerPopupOnMapClick = false;
+            if (this.dialog) {
+                this.dialog.close(true);
             }
+            this.popupCleanup();
+            this.getMapModule().getMapEl().removeClass('cursor-crosshair');
             if (!selectDefault) {
                 return;
             }
             // ask toolbar to select default tool if available
-            var toolbarRequest = Oskari.requestBuilder('Toolbar.SelectToolButtonRequest');
+            const toolbarRequest = Oskari.requestBuilder('Toolbar.SelectToolButtonRequest');
             if (toolbarRequest) {
-                sandbox.request(me, toolbarRequest());
-            }
-        },
-
-        /**
-         * Adds an array of markers to the map
-         * @param markers
-         */
-        addMapMarkers: function (markers) {
-            var i;
-            for (i = 0; i < markers.length; i += 1) {
-                this.addMapMarker(markers[i], null, true);
+                this.getSandbox().request(this, toolbarRequest());
             }
         },
-
-        __getSanitizedMarker: function (markerData, id) {
+        _getSanitizedMarker: function (markerData, markerId) {
+            const { x, y } = markerData;
             // Validation: coordinates are needed
-            if ((typeof markerData.x === 'undefined') || (typeof markerData.y === 'undefined')) {
-                this.getSandbox().printWarn('Undefined coordinate in', markerData);
-                return null;
+            if ((typeof x === 'undefined') || (typeof y === 'undefined')) {
+                this.log.warn('Undefined coordinate in', markerData);
+                return {};
             }
             // Remove null values to get defaults
             Object.keys(markerData).forEach(key => markerData[key] === null && delete markerData[key]);
-            // Combine default values with given values
-            var data = jQuery.extend(true, {}, this._defaultData, markerData);
+
+            if (markerData.color && markerData.color.charAt(0) === '#') {
+                markerData.color = markerData.color.substring(1);
```

Comment:
This modifies the input parameter and might cause problems later on. Maybe use the spread operator at the start of the function to generate a copy of the input before modifying it rather than at the return statement?